sdk/core/azure-core/src/main/java/com/azure/core/http/ProxyOptions.java

```diff
@@ -168,7 +156,16 @@ public String getNonProxyHosts() {
      * @throws IllegalArgumentException If {@code configuration} is {@link Configuration#NONE}.
      */
     public static ProxyOptions fromConfiguration(Configuration configuration) {
-        return fromConfiguration(configuration, false);
+        if (configuration == Configuration.NONE) {
```

Comment:
this is an existing pattern we use everywhere,  But I don't see how `Configuration.NONE` is different than null, I'll chat with @alzimmermsft once he's back to understand if we can improve something here.

Reply:
I'm trying to recall why `Context.NONE` throws instead of returning a null proxy but I can't remember why, and I agree this behavior seems odd. I wonder if we can change this to return null instead without any major consequences.

Reply:
The difference between `NONE` and `null` configuration I realized: 
- `NONE` is just an empty config and should always lead to `null` outcome
- `null` configuration means implicit (enviroment) configuration

I believe it's not breaking to return `null` instead of throwing:
- The underlying method `attemptToLoadProxy` returns `null`  when no proxy configuration is found, i.e. callers should expect to get `null`
- stop throwing an exception is usually considered non-breaking. It's hard to imagine a case where someone relies on this exception being thrown 

I.e. for `ProxyOptions`, it's not breaking to remove the exception and we don't use it in other places in azure-core. 

This is also consistent with what we (in general) do in client libraries:

https://github.com/Azure/azure-sdk-for-java/blob/b131f86920a386fd7191da6025b2901ffb6c4bfa/sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java#L456-L465